ge/org.osate.ge/src/org/osate/ge/query/ExectableQuery.java

```diff
@@ -25,52 +25,88 @@
 
 import java.util.List;
 import java.util.Optional;
-import java.util.function.Function;
+import java.util.function.UnaryOperator;
 
 import org.osate.ge.BusinessObjectContext;
 import org.osate.ge.internal.query.DefaultQuery;
 import org.osate.ge.internal.query.QueryRunner;
 import org.osate.ge.internal.query.RootQuery;
+import org.osate.ge.services.QueryService;
 
 /**
+ * Represents a query which can be executed by @link QueryService} to retrieve results
  * @noextend This class is not intended to be subclassed by clients.
- *
+ * @since 3.0
  */
-public class StandaloneQuery {
-	private final DefaultQuery rootQuery = new RootQuery(() -> this.rootNode);
-	private BusinessObjectContext rootNode;
-	private final DefaultQuery query;
+public final class ExectableQuery<T> {
```

Comment:
Should this be ExecutableQuery?

Reply:
Should be fixed now. Thanks for catching this. It took me a minute to figure out what your comment meant.